### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Vindexus**

### CODE REVIEW

File: src/js/events.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    return transformEvent(r)
+  })
+  return results  
+}
+
+function transformEvent (i) {
+  i.upcoming = new Date(i.startDate) > new Date()
+  i.dateString = formatDate(i.startDate)
+  i.date = formatDateJSON(i.startDate)
+  i.month = 'Sep'
```
Comment (lostlevels) [inline src/js/events.js]: Is there reason why this is `Sep`?

Comment (Vindexus) [inline src/js/events.js]: Nope, just left over testing.

File: src/js/events.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  return obj
+}
+
+function subscribeEventsEmailOptIn (e, el) {
+  var data = getTargetDataSet(el, true, true)
+  if(!data.googleMapsPlaceId) {
+    alert('Please enter your location')
+    return
+  }
+  data['emailOptIns[eventsNearMe]'] = true
```
Comment (lostlevels) [inline src/js/events.js]: Not sure this is what you wanted.

File: src/js/events.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    qs.email = data.email
+  }
+  return go('/sign-up?' + objectToQueryString(qs))
+}
+
+function getUpcomingEventsQueryObject (options) {
+  var page = options.hasOwnProperty('page') ? parseInt(options.page) : 1
+  if(!page || page < 1) {
+    page = 1
+  }
```
Comment (lostlevels) [inline src/js/events.js]: Seems superfluous, `var page = parseInt(options.page) || 1` would do

File: src/js/signin.js [hunk 4]
```diff
@@ -141,20 +146,44 @@ function getSignInBuying () {
...
-  return {
+  obj = {
     countries: getAccountCountries(),
     buying: buying,
     redirectTo: encodeURIComponent(redirectTo)
   }
+
+  var qo = queryStringToObject(window.location.search)
+
+  if(qo.email && qo.email && qo.email != 'undefined') {
```
Comment (lostlevels) [inline src/js/signin.js]: `qo.email` twice?

File: src/js/signin.js [hunk 5]
```diff
@@ -141,20 +146,44 @@ function getSignInBuying () {
...
+  }
+
+  return obj
 }
 
 function mapConfirmSignup () {
   var obj = queryStringToObject(window.location.search)
   if (!Object.keys(obj).length) return
   obj.countries = getAccountCountries()
+  if(obj.email == 'undefined') {
```
Comment (lostlevels) [inline src/js/signin.js]: Same here

File: src/js/signin.js (hunk 4 above)
Comment (Vindexus) [inline src/js/signin.js]: Double qo.email is typo. When I signed up through Facebook the email it provided was "undefined", but I think that was a different place in the code.

File: src/js/signin.js (hunk 5 above)
Comment (Vindexus) [inline src/js/signin.js]: Here it definitely happens.

File: src/js/events.js (hunk 2 above)
Comment (Vindexus) [inline src/js/events.js]: This is how [] style fields are formatted with declare so I did it this way to stay consistent. data.emailOpts.eventsNearMe = true would be nicer, but getTargetDataSel doesn't do subobjects.

**Pull request merged**